gocron.go

```diff
@@ -115,20 +115,22 @@ func (j *Job) run() (result []reflect.Value, err error) {
 		}()
 	}
 
-	f := reflect.ValueOf(j.funcs[j.jobFunc])
-	params := j.fparams[j.jobFunc]
+	j.lastRun = time.Now()
+	result, err = callJobFuncWithParams(j.funcs[j.jobFunc], j.fparams[j.jobFunc])
+	j.scheduleNextRun()
+	return
+}
+
+func callJobFuncWithParams(jobFunc interface{}, params []interface{}) ([]reflect.Value, error) {
+	f := reflect.ValueOf(jobFunc)
 	if len(params) != f.Type().NumIn() {
```

Comment:
This logic is no longer accurate for `DoSafely` as a result of this PR. When you wrap the jobFunc for safely, f becomes the jobFunc and the check is wrong now. It will check the wrapped function not the passed function. So `f.Type().NumIn()` will ALWAYS return 1 regardless of the jobFunc passed into DoSafely. 

Results in any function that does not have EXACTLY 1 parameter failing this check and not running.

Reply:
@tiger5226 we will be deprecating DoSafely in a few days as we will stop bubbling panics up the chain, thus there is no need for it anymore (it'll be the default behavior of Do)